### ISSUE TITLE
TFLint 0.40.0 not compatible with Azurerm plugin

### ISSUE THREAD

**Issue by danielkimuipath** [terraform-linters/tflint#1506]
## Introduction

Hi! there is an issue with the recent version of TfLint(0.40.0) with azurerm provider (2.98.X) version of azurerm provider. We have a line to install the latest master version of tflint and then after 0.40.0 version was released, pipelines start to fail due to tflint failure on initializing.. Can someone take a look at this issue? thank you! 

Issue from the pipeline:

```hcl
Current tflint version
TFLint version 0.40.0
Plugin `azurerm` is already installed
Failed to initialize plugins; TFLint is not compatible with this version of the "azurerm" plugin. A newer TFLint or plugin version may be required. Plugin version: 10, Client versions: [11]

```
Version of the azure rm provider

```
    azurerm = {
      source = "hashicorp/azurerm"
      version = ">=2.98.0, < 3.0.0"
    }
```

## Expected Behavior
This should pass the installation of the TfLint of 0.40.0 version

## Actual behavior
Installation of the TfLint fails

## Step to Reproduce

Install TFlint of 0.40.0 and azurerm of v2.98.1 version and run tflint command

## Additional Context
The previous version of TfLint works successfully - `0.39.3`

```
====================================================
Unpacking /tmp/tflint.EyFKEiWwRj/tflint.zip ...
Archive:  /tmp/tflint.EyFKEiWwRj/tflint.zip
  inflating: /tmp/tflint.EyFKEiWwRj/tflint  
Installing /tmp/tflint.EyFKEiWwRj/tflint to /usr/local/bin/ ...
'/tmp/tflint.EyFKEiWwRj/tflint' -> '/usr/local/bin/tflint'
Cleaning temporary downloaded files directory /tmp/tflint.EyFKEiWwRj ...


====================================================
Current tflint version
TFLint version 0.39.3
Installing `azurerm` plugin...
Installed `azurerm` (source: github.com/terraform-linters/tflint-ruleset-azurerm, version: 0.17.0)

```

**Comment by bendrucker** [comment]
Unrelated to the provider version you're using. The plugin will need to be updated:

https://github.com/terraform-linters/tflint-ruleset-azurerm/releases/tag/v0.18.0
